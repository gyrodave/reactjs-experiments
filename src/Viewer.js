import React from 'react';
import injectSheet from 'react-jss'

const styles = {
  viewer: {
    left: '0',
    'text-align': 'left'
  },
};

const Viewer = ({classes, text }) => {
  return (
    <div>
      {text.map(e => <div className={classes.viewer} key={e}>{e}</div>)}
    </div>
  );
};

export default injectSheet(styles)(Viewer);
