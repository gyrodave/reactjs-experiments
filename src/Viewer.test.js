import React from 'react';
import Viewer from './Viewer';
import { shallow } from 'enzyme';

it('tests Viewer Component', () => {
  const text = ['my super test', 'my super test 2'];
  const wrapper = shallow(<Viewer text={text} />).dive();

  expect(wrapper.children().length).toBe(text.length);
  wrapper.children().forEach((elem, index) => {
    expect(elem.text()).toBe(text[index]);
  });
});
