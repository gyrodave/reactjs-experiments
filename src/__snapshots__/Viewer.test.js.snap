// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`tests Viewer Component 1`] = `
<div>
  <div
    className="viewer-0-0"
  >
    my super test
  </div>
  <div
    className="viewer-0-0"
  >
    my super test 2
  </div>
</div>
`;
